src/server/scripts/Events/midsummer.cpp

```diff
@@ -49,6 +68,319 @@ class go_midsummer_bonfire : public GameObjectScript
+                {
+                    _spellFocus->DespawnOrUnsummon();
+                    _spellFocus = nullptr;
+                }
+
+                me->CastSpell(me, SPELL_STAMP_OUT_BONFIRE_ART_KIT, true);
+                UpdateBonfireBlessingBuffs();
+                break;
+            default:
+                break;
+        }
+    }
+
+    void UpdateBonfireBlessingBuffs()
+    {
+        if ((_type != BONFIRE_TYPE_ALLIANCE) && (_type != BONFIRE_TYPE_HORDE))
+            return;
+
+        me->GetMap()->DoForAllPlayers([&](Player* p)
+            {
+                if ((p->GetZoneId() == me->GetZoneId()) && (p->GetTeamId() == _teamId))
+                {
+                    if (_isStampedOut)
+                    {
+                        if (*_isStampedOut)
+                        {
+                            p->RemoveAurasDueToSpell(SPELL_BONFIRES_BLESSING);
+                        }
+                        else
+                        {
+                            if (!p->HasAura(SPELL_BONFIRES_BLESSING))
+                                p->CastSpell(p, SPELL_BONFIRES_BLESSING, true);
+                        }
+                    }
+                }
+            });
+    }
+
+    bool InitBonfire()
+    {
+        _type = BONFIRE_TYPE_NONE;
+        _teamId = TEAM_NEUTRAL;
+
+        for (uint32 i = 0; (i < COUNT_GO_BONFIRE_ALLIANCE) && (_type == BONFIRE_TYPE_NONE); i++)
+        {
+            if (_bonfire = me->FindNearestGameObject(GoBonfireAlliance[i], 10.0f))
+            {
+                _type = BONFIRE_TYPE_ALLIANCE;
+                _teamId = TEAM_ALLIANCE;
+            }
+        }
+
+        for (uint32 i = 0; (i < COUNT_GO_BONFIRE_HORDE) && (_type == BONFIRE_TYPE_NONE); i++)
+        {
+            if (_bonfire = me->FindNearestGameObject(GoBonfireHorde[i], 10.0f))
+            {
+                _type = BONFIRE_TYPE_HORDE;
+                _teamId = TEAM_HORDE;
+            }
+        }
+
+        for (uint32 i = 0; (i < COUNT_GO_BONFIRE_CITY) && (_type == BONFIRE_TYPE_NONE); i++)
+        {
+            if (_bonfire = me->FindNearestGameObject(GoBonfireCity[i], 10.0f))
+            {
+                _type = BONFIRE_TYPE_CITY;
+                return true;
+            }
+        }
+
+        if ((_type == BONFIRE_TYPE_NONE) && (_bonfire = me->FindNearestGameObject(GO_AHUNE_BONFIRE, 10.0f)))
+        {
+            _type = BONFIRE_TYPE_AHUNE;
+        }
+
+        if (_type == BONFIRE_TYPE_NONE)
+            return false;
+
+        auto itr = BonfireStateStore.find(std::make_tuple(me->GetMapId(), me->GetZoneId(), _teamId));
+        if ((itr != BonfireStateStore.end()) && (itr->second))
+        {
+            _isStampedOut = itr->second;
+        }
+        else
+        {
+            LOG_ERROR("scripts.midsummer", "NPC {} (GUID{}) in map {}, zone {} with teamId {} can't locate its entry within BonfireStateStore", me->GetGUID().GetEntry(), me->GetSpawnId(), me->GetMapId(), me->GetZoneId(), _teamId);
```

Comment:
is scripts.midsummer added in the conf file?

Reply:
No 👀 maybe I'll just drop the error message for now

Reply:
just add it to the conf file. LOG_ERROR will still send iirc but we still want it in conf